packages/react-components/components/PhoneNumberInput.tsx

```diff
@@ -69,40 +69,56 @@ export default class PhoneNumberInput extends React.Component<Props, State> {
     }
   }
 
+  async getPhoneNumberFromNativePicker() {
+    // linter doesn't allow following code without splitting this code in two lines
+    const number = await SmsRetriever.requestPhoneNumber()
+    return number
+  }
+
   async triggerPhoneNumberRequestAndroid() {
     try {
-      const phone = await SmsRetriever.requestPhoneNumber()
-      const phoneNumber = parsePhoneNumber(phone, '')
+      let phone
+      try {
+        phone = await this.getPhoneNumberFromNativePicker()
+      } catch (error) {
+        console.info(
+          `${TAG}/triggerPhoneNumberRequestAndroid`,
+          'Could not request phone. This might be thrown if the user dismissed the modal',
+          error
+        )
+        return
+      }
 
+      const phoneNumber = parsePhoneNumber(phone, '')
       if (!phoneNumber) {
         return
       }
-
       this.setState({ phoneNumber: phoneNumber.displayNumber.toString() })
 
-      if (phoneNumber.countryCode) {
-        // TODO known issue, the country code is not enough to
-        // get a country, e.g. +1 could be USA or Canada
-        const displayName = this.state.countries.getCountryByPhoneCountryCode(
-          '+' + phoneNumber.countryCode.toString()
-        ).displayName
+      const regionCode = phoneNumber.regionCode
 
+      if (regionCode) {
+        const displayName = this.state.countries.getCountryByCode(regionCode).displayName
         this.onChangeCountryQuery(displayName)
       }
     } catch (error) {
-      console.error(`${TAG}/triggerPhoneNumberRequestAndroid`, 'Could not request phone', error)
+      console.info(`${TAG}/triggerPhoneNumberRequestAndroid`, 'Could not request phone', error)
     }
   }
 
   async triggerPhoneNumberRequest() {
+    if (this.state.phoneNumber || this.state.countryQuery) {
```

Comment:
Could this logic be in `onCountryFocus` so that `triggerPhoneNumberRequest` is only called if necessary?